persist/sqlite/consensus.go

```diff
@@ -649,28 +649,17 @@ func addEvents(tx *txn, scDBIds map[types.SiacoinOutputID]int64, fcDBIds map[exp
 	}
 	defer foundationSubsidyEventStmt.Close()
 
-	var buf bytes.Buffer
-	enc := json.NewEncoder(&buf)
+	// Insert/update host announcements if there are any.
+	// The loop below this one checks if we've already seen an event
+	// transaction with the same ID before it decides to insert it.  Here we
+	// want to insert regardless to update the last_announcement field in the
+	// event that we've seen this host before.  If the transaction pays a fee
+	// which most real transactions do then this unnecessary because the txn ID
+	// will be unique regardless due to the siacoin input but this is more
+	// technically correct and makes testing easier.
 	for _, event := range events {
```

Comment:
It would be simpler and more efficient to add announcements to the UpdateState with a helper struct and add them to the database directly. You end up doing many slightly different loops over the same data this way when very few events will actually have an announcement. It would also reinforce that host announcements should not update all of the fields.